sseclient/__init__.py

```diff
@@ -45,12 +46,16 @@ def _read(self):
         to correctly stitch together consecutive response chunks and find the
         SSE delimiter (empty new line) to yield full, correct event chunks."""
         data = b''
+        cursor = 0
         for chunk in self._event_source:
-            for line in chunk.splitlines(True):
-                if not line.strip():
-                    yield data
-                    data = b''
-                data += line
+            data += chunk
+            delim = max(map(lambda s: (data.find(s, cursor), len(s)), ('\r\n\r\n', '\r\r', '\n\n')))
```

Comment:
@mpetazzoni apologies for the delayed response - `max()` keys on the first element of the tuple by default. So `max((0, 4), (1, 2), (6, 3))` would return `(6, 3)`.